### PR TITLE
Torpedostore bugfixes

### PR DESCRIPTION
**Pull request by attilakillin**
Fix three separate issues highlighted by SonarLint in the TorpedoStore class.
Each fix is in a separate commit.

### CODE REVIEW

**Comment by attilakillin** [comment]
Replaced Random with SecureRandom too, as the SonarCloud checks failed with the default, unsafe implementation.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -28,20 +28,21 @@ public TorpedoStore(int numberOfTorpedos){
     }
   }
 
+  private SecureRandom generator = new SecureRandom();
```
Comment (karsazoltan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Add SecureRandom to generate random numbers fire function

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -28,20 +28,21 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (karsazoltan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: decrease torpedoCount succesfull firing

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -28,20 +28,21 @@ public TorpedoStore(int numberOfTorpedos){
     }
   }
 
+  private SecureRandom generator = new SecureRandom();
+
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (karsazoltan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: add throw op

**Review by karsazoltan — COMMENTED** [review]
Check bugfixes

**Review by karsazoltan — COMMENTED** [review]
Static generator

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (karsazoltan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:31]: static generator

**Review by karsazoltan — APPROVED** [review]

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (karsazoltan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: set static generator

**Review by karsazoltan — CHANGES_REQUESTED** [review]
add static

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -28,20 +28,21 @@ public TorpedoStore(int numberOfTorpedos){
     }
   }
 
+  private static SecureRandom generator = new SecureRandom();
```
Comment (karsazoltan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:31]: good

**Review by karsazoltan — APPROVED** [review]

**Pull request merged**